decrypt.go

```diff
@@ -41,6 +74,33 @@ func (p7 *PKCS7) Decrypt(cert *x509.Certificate, pkey crypto.PrivateKey) ([]byte
 	return nil, ErrUnsupportedAlgorithm
 }
 
+func getHashFuncForKeyEncryptionAlgorithm(keyEncryptionAlgorithm pkix.AlgorithmIdentifier) (crypto.Hash, error) {
+	invalidHashFunc := crypto.Hash(0)
+	params := new(rsaOAEPAlgorithmParameters)
+	var rest []byte
+	rest, err := asn1.Unmarshal(keyEncryptionAlgorithm.Parameters.FullBytes, params)
+	if err != nil {
+		return invalidHashFunc, fmt.Errorf("failed unmarshaling key encryption algorithm parameters: %s", err.Error())
+	}
+	if len(rest) != 0 {
+		return invalidHashFunc, errors.New("trailing data after RSAES-OAEP parameters")
+	}
+	// The default is SHA-1.
+	if params.HashFunc.Algorithm == nil || params.HashFunc.Algorithm.Equal(OIDDigestAlgorithmSHA1) {
```

Comment:
this is why we have `switch` statements.

Reply:
Fixed in b1fb0e54346f2f8b218ca002427fe0eff22564f9